### PR TITLE
add conv2d dw class

### PR DESCRIPTION
**Pull request by andrewstanfordjason**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
+1

File: test/gtests/src/op/test_AvgPool2d.cpp [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+       new FrameSequence({new Apply(MakeUnpaddedDepthwise),
+                          new Apply(MakePaddedDepthwise)})});
+}
+
+/**
+ * Test Case -- Compare output of  AvgPool2d_Generic to our reference
+ * implementation
+ */
+TEST(AvgPool2d_Generic_Test, CompareWithReference) {
+  /*
```
Comment (ghost) [inline test/gtests/src/op/test_AvgPool2d.cpp:98]: Should these tests be enabled?

File: test/gtests/src/op/test_AvgPool2d.cpp [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+  std::cout << "Count: " << total_iters << std::endl;
+  */
+}
+
+/**
+ * Test Case -- Compare output of  AvgPool2d_Valid to our reference
+ * implementation
+ */
+TEST(AvgPool2d_Valid_Test, CompareWithReference) {
+  /*
```
Comment (ghost) [inline test/gtests/src/op/test_AvgPool2d.cpp:205]: Should these tests be enabled?

File: test/gtests/src/op/test_MaxPool2d.cpp [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+       new FrameSequence({new Apply(MakeUnpaddedDepthwise),
+                          new Apply(MakePaddedDepthwise)})});
+}
+
+/**
+ * Test Case -- Compare output of  MaxPool2d_Generic to our reference
+ * implementation
+ */
+TEST(MaxPool2d_Generic_Test, CompareWithReference) {
+  /*
```
Comment (ghost) [inline test/gtests/src/op/test_MaxPool2d.cpp:97]: Should these tests be enabled?

File: test/gtests/src/op/test_MaxPool2d.cpp [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+  std::cout << "Count: " << total_iters << std::endl;
+  */
+}
+
+/**
+ * Test Case -- Compare output of  MaxPool2d_Valid to our reference
+ * implementation
+ */
+TEST(MaxPool2d_Valid_Test, CompareWithReference) {
+  /*
```
Comment (ghost) [inline test/gtests/src/op/test_MaxPool2d.cpp:187]: Should these tests be enabled?

**Comment by andrewstanfordjason** [comment]
Sorry, it's not ready yet, I've just added the tag. I thought it best to add the classes, which seem good, with the tests, which, right now, fail the asan. I'll be fixing them in a future PR but max/avg pool are lower priority

**Comment by ghost** [comment]
No worries, that's alright!

**Pull request merged**